packages/sitecore-jss-cli/src/scripts/extract-components.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  fetchBearerToken,
+  resolveComponentImportFiles,
+  sendCode,
+} from '@sitecore-jss/sitecore-jss-dev-tools';
+import chalk from 'chalk';
+import path from 'path';
+import { Argv } from 'yargs';
+
+/**
+ * @param {Argv} yargs
+ */
+export function args(yargs: Argv) {
+  return yargs
+    .option('environment', {
+      requiresArg: false,
+      type: 'string',
+      describe: 'Environment to authenticate into. Default: prod',
+    })
+    .option('appFolder', {
+      requiresArg: false,
+      type: 'string',
+      describe: 'Path to app folder to get components from. Default: current folder',
+    });
+}
+
+/**
+ * @param {Argv} yargs
+ */
+export default function builder(yargs: Argv) {
+  return yargs.command(
+    'extract-components',
+    'Reads component files and posts their code for AI learning',
+    args,
+    handler
+  );
+}
+
+/**
+ * Handler for the extract-component API command
+ * Reads imports from the componentBuilder.ts file and posts the code to the mesh endpoint
+ * @param {object} args - The arguments passed to the command, with optional appName string
+ * @returns {Promise<void>} void
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export async function handler(args: any) {
```

Comment:
We shouldn't use `any` if it's possible to avoid
```suggestion
interface CommandArgs {
   appFolder: string;
   ...
}
...
export async function handler(args: CommandArgs) {
```

Reply:
yargs `builder` does not compile if strict type is used
Besides, that's the approach we use in other commands already.